pinot-core/src/main/java/org/apache/pinot/core/auth/BasicAuthPrincipal.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.auth;
+
+import java.util.Set;
+
+
+/**
+ * Container object for basic auth principal
+ */
+public class BasicAuthPrincipal {
+  private final String _name;
+  private final String _token;
+  private final Set<String> _tables;
+  private final Set<String> _permissions;
```

Comment:
should be `String` to retain extensibility of AccessControl implementations

Reply:
I think it should be "AccessType" because AccessControl.hasAccess() takes an AccessType object as its argument. Implementations of AccessControl can't change AccessType.

Reply:
(Double-posting here for context) I agree with this from the controller side, however, AccessType is a controller concept only - broker implementations may specify other types of access in an extensible way. One could imagine limiting access to PII data and similar